Jenkinsfile

```diff
@@ -1,59 +1,100 @@
-                '''
+            steps {
+                sh 'xvfb-run ./gradlew \
+                    -PbuildNumber=$BUILD_NUMBER \
+                    -PbuildAccess=$BUILD_ACCESS \
+                    -PbuildTag=$TAG_NAME \
+                    -PartifactoryUrl=$ARTIFACTORY_URL \
+                    -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR \
+                    -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW \
+                    -PtestrailHost=$TESTRAIL_HOST \
+                    -PtestrailUser=$TESTRAIL_CREDENTIALS_USR \
+                    -PtestrailPassword=$TESTRAIL_CREDENTIALS_PSW \
+                    -PtestrailSuiteId=$TESTRAIL_SUITE_ID \
+                    -PadditionalTestArguments=$ADDITIONAL_TEST_ARGUMENTS \
+                    --gradle-user-home GRADLE_USER_HOME \
+                    --continue --info --stacktrace \
+                    check testrailPublish'
+            }
+        }
+        stage('Publish Snapshot') {
+            when { branch 'develop' }
+            steps {
+                sh './gradlew \
+                -PbuildNumber=1000 \
+                -PbuildAccess=$BUILD_ACCESS \
+                -PbuildTag=$TAG_NAME \
+                -PartifactoryUrl=$ARTIFACTORY_URL \
+                -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR \
+                -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW \
+                -PartifactoryRepository=maven-libs-snapshot-local \
+                --gradle-user-home GRADLE_USER_HOME \
+                --continue --info --stacktrace \
+                artifactoryPublish'
             }
         }
-        stage('Publish') {
-            mdk_builder.inside{
-                sh '''
-                export GRADLE_USER_HOME=$(pwd)/.gradle
-                if [ -z $GIT_TAG ]; then ARTIFACTORY_REPOSITORY="maven-libs-snapshot-local"; else ARTIFACTORY_REPOSITORY="maven-libs-release-local"; fi
-                ./gradlew -PbuildNumber=$BUILD_NUMBER -PbuildAccess=$BUILD_ACCESS -PbuildTag=$GIT_TAG -PartifactoryUrl=$ARTIFACTORY_URL -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW -PartifactoryRepository=$ARTIFACTORY_REPOSITORY --gradle-user-home GRADLE_USER_HOME --continue --info --stacktrace artifactoryPublish
-                '''
+
+        stage('Publish Release') {
+            when { allOf { branch 'master';  tag "" } }
+            steps {
+                sh './gradlew \
+                -PbuildNumber=$BUILD_NUMBER \
+                -PbuildAccess=$BUILD_ACCESS \
+                -PbuildTag=$TAG_NAME \
+                -PartifactoryUrl=$ARTIFACTORY_URL \
+                -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR \
+                -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW \
+                -PartifactoryRepository=maven-libs-release-local \
```

Comment:
See above.

Reply:
The Aritfactory Repository used to be set with an If-Else based on whether a TAG was set. Now, since I'm conditionally running the snapshot or release based on TAG_NAME. There is not a credential right now for this repo string, but I could add one if you expect it to ever change.

Reply:
Nah I must've been confusing it with the CircleCI build. Resolved.